### PR TITLE
accessibility: fixes issue/#2075, issue/#2084 and issue/#2078

### PR DESCRIPTION
**Pull request by oliverfoster**
[#2075](https://github.com/adaptlearning/adapt_framework/issues/2075)
* Removed a11y_text
* Removed unnecessary roles

[#2084](https://github.com/adaptlearning/adapt_framework/issues/2084)
* Removed component description

[#2078](https://github.com/adaptlearning/adapt_framework/issues/2078) 
* Removed accessibility mode

[#2165](https://github.com/adaptlearning/adapt_framework/issues/2165)
* Updated component description 

[#2190](https://github.com/adaptlearning/adapt_framework/issues/2190) mejs fixes
* Override / removed keyActions
* Fixed default seek functions
* Removed unnecessary aria-controls and role attributes
* Added aria-hidden to video tag to stop the screenreader going into forms mode
* Fixed styling to prevent ios iframe from scaling

[#2191](https://github.com/adaptlearning/adapt_framework/issues/2191)
* Added skip to transcript button

List of outstanding PRs: https://github.com/adaptlearning/adapt_framework/issues/2206.

### CODE REVIEW

File: js/adapt-contrib-media.js [hunk 1]
```diff
@@ -28,6 +28,18 @@ define([
         }
     }
 
+    _.extend(mejs.MepDefaults, {
```
Comment (moloko) [inline js/adapt-contrib-media.js]: could you add a comment explaining why this is necessary?

File: properties.schema [hunk 2]
```diff
@@ -7,15 +7,15 @@
     "ariaRegion": {
       "type": "string",
       "required": true,
-      "default": "This is a media player component. Select the play / pause button to watch or listen.",
+      "default": "Media player{{#any _transcript._inlineTranscript _transcript._externalTranscript}} and transcript{{/any}}.",
```
Comment (brian-learningpool) [inline properties.schema]: What's going on here?

Comment (oliverfoster) [reply properties.schema]: if any of those attributes has a truthy add `and transcript` to the end.

**Comment by oliverfoster** [comment]
one more +1 need here @tomgreenfield ?

File: js/adapt-contrib-media.js [hunk 3]
```diff
@@ -163,6 +175,12 @@ define([
             }
         },
 
+        cleanUpPlayer: function() {
+            this.$('.media-widget').children('.mejs-offscreen').remove();
+            this.$('[role=application]').removeAttr('role').removeAttr("tabindex");
```
Comment (tomgreenfield) [inline js/adapt-contrib-media.js]:
```js
this.$('[role=application]').removeAttr('role tabindex');
```

File: js/adapt-contrib-media.js [hunk 4]
```diff
@@ -391,25 +410,28 @@ define([
             this.$('audio, video').width(this.$('.component-widget').width());
         },
 
-        onAccessibilityToggle: function() {
-           this.showControls();
+        onSkipToTranscript: function() {
+            this.$('.media-transcript-container button').a11y_focus();
```
Comment (tomgreenfield) [inline js/adapt-contrib-media.js]:
```js
this.$('.media-transcript-container').find('button').a11y_focus();
```

File: less/media.less [hunk 5]
```diff
@@ -26,6 +26,7 @@
 
     .mejs-container {
         background-color: transparent;
+        width: 100% !important;
```
Comment (tomgreenfield) [inline less/media.less]: Comment explaining use of `!important`?

Comment (oliverfoster) [reply less/media.less]: mejs tried to set a pixel width, but it's always 100% of the container.

Comment (oliverfoster) [reply less/media.less]: that's a problem on ios, when you rotate it assumes that the container is a fixed size and tries to scale the browser.
having a percentage measure means that safari knows that it'll be resized on orientation change and avoids the scaling code.

File: js/adapt-contrib-media.js (hunk 4 above)
Comment (oliverfoster) [reply js/adapt-contrib-media.js]: why?

File: less/media.less (hunk 5 above)
Comment (tomgreenfield) [reply less/media.less]: Sorry, meant a quick, written comment in the Less.

**Pull request merged**